### PR TITLE
Lp deletebudgetlineitem

### PR DESCRIPTION
**Pull request by ldphilli**

### CODE REVIEW

File: Client/ViewModels/BudgetLineItemViewModel.cs [hunk 1]
```diff
@@ -12,16 +11,26 @@ public class BudgetLineItemViewModel
...
         {
             ApiClient = apiClient;
+            appState = applicationState;
+            nvm = navManager;
             bliModel = new BudgetLineItemModel();
             bliModel.Id = Guid.NewGuid().ToString();
-            appState = applicationState;
         }
 
+        // public async Task CreateBudgetLineItem()
```
Comment (tdashworth) [inline Client/ViewModels/BudgetLineItemViewModel.cs:25]: Commented out. Could be removed

File: Client/Pages/Budget/EditBudgetLineItem.razor [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        </RadzenCard>
+    </div>
+    <RadzenButton class="col-xl-3" Click="@(() => BudgetLineViewModel.UpdateBudgetLineItem(ApplicationState.BudgetLineItemModel))" Text="Save" ButtonStyle="ButtonStyle.Primary"></RadzenButton>
+    <RadzenButton class="col-xl-3" Click="@(() => BudgetLineViewModel.DeleteBudgetLineItem(ApplicationState.BudgetLineItemModel))" Text="Delete" ButtonStyle="ButtonStyle.Primary"></RadzenButton>
+</div>
+
+
+@code {
+
+    protected override async Task OnInitializedAsync() {
```
Comment (tdashworth) [inline Client/Pages/Budget/EditBudgetLineItem.razor:51]: Should handle if the app state isn't set (`null`)

File: Client/ViewModels/MonthlyBudgetViewModel.razor.cs [hunk 3]
```diff
@@ -96,6 +103,7 @@ public MonthlyBudgetViewModel(HttpClient apiClient, NavigationManager navManager
             appState.monthlyBudgetModel.Remaining = Math.Round(appState.monthlyBudgetModel.Income - appState.monthlyBudgetModel.Outgoing, 2);
             await monthlyBudgetService.UpdateMonthlyBudgetForUser(monthlyBudgetModel);
             //ReloadBudgetForSelectedMonth();
+            //nvm.NavigateTo("/budget");
```
Comment (tdashworth) [inline Client/ViewModels/MonthlyBudgetViewModel.razor.cs:106]: Commented out

File: Client/ViewModels/MonthlyBudgetViewModel.razor.cs [hunk 4]
```diff
@@ -52,6 +56,9 @@ public MonthlyBudgetViewModel(HttpClient apiClient, NavigationManager navManager
 
             appState.monthlyBudgetModel = GetSpecificMonthlyBudget();
 
+            expenditureCategories = appState.monthlyBudgetModel.expenditureCategories;
+            Console.WriteLine(expenditureCategories.First().CategoryName);
```
Comment (tdashworth) [inline Client/ViewModels/MonthlyBudgetViewModel.razor.cs:60]: I would consider removing any of these if not required anymore

**Review by tdashworth — COMMENTED** [review]
Some comments (sorry that they are short and sharp)

File: Client/Pages/Budget/EditBudgetLineItem.razor (hunk 2 above)
Comment (ldphilli) [reply Client/Pages/Budget/EditBudgetLineItem.razor:51]: I wasn't sure how best to handle it so I declared a new model so it doesn't throw errors

**Pull request merged**